virttest/libvirt_xml/vm_xml.py

```diff
@@ -554,21 +554,30 @@ def del_controller(self, controller_type=None):
 
         :return: None if deleting all controllers
         """
-        all_controllers = self.xmltreefile.findall("devices/controller")
-        del_controllers = []
-        for controller in all_controllers:
-            if controller.get("type") != controller_type:
-                continue
-            del_controllers.append(controller)
-
         # no seclabel tag found in xml.
+        del_controllers = self.get_controllers()
```

Comment:
we have to pass the `controller type` else it would break for non default value
```python
del_controllers = self.get_controllers(controller_type=controller_type)
```

Reply:
Fixed, thanks